src/main/java/org/sonar/ucfg/UCFGtoProtobuf.java

```diff
@@ -0,0 +1,158 @@
+/*
+ * sonar-ucfg
+ * Copyright (C) 2018-2018 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.ucfg;
+
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.Map;
+import java.util.stream.Collectors;
+import javax.annotation.Nullable;
+import org.sonar.ucfg.protobuf.Ucfg;
+
+public final class UCFGtoProtobuf {
+
+  private UCFGtoProtobuf() {
+    // empty constructor
+  }
+
+  public static void toProtobufFile(UCFG ucfg, String filename) {
+    Ucfg.UCFG protobufUCFG = Ucfg.UCFG.newBuilder()
+      .setMethodId(ucfg.methodId())
+      .setLocation(toProtobuf(ucfg.location()))
+      .addAllParameters(ucfg.parameters().stream().map(Expression.Variable::id).collect(Collectors.toList()))
+      .addAllUcfgEntries(ucfg.entryBlocks().stream().map(BasicBlock::label).map(Label::id).collect(Collectors.toList()))
+      .addAllBasicBlocks(ucfg.basicBlocks().values().stream().map(UCFGtoProtobuf::toProtobuf).collect(Collectors.toList()))
+      .build();
+
+    try {
+      protobufUCFG.writeTo(new FileOutputStream(filename));
+    } catch (IOException e) {
+      e.printStackTrace();
```

Comment:
![Vulnerability](https://sonarsource.github.io/sonar-github/vulnerability.png 'Vulnerability') Vulnerability: Use a logger to log this exception. ([squid:S1148](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1148))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.ucfg%3Asonar-ucfg&pullRequest=1&issues=AWJJDDis5ZbnBykCSQD4&open=AWJJDDis5ZbnBykCSQD4)

[ISSUE_KEY]: AWJJDDis5ZbnBykCSQD4